plugins/bazel/cargo/Cargo.toml

```diff
@@ -3,13 +3,13 @@ name = "service-extensions"
 version = "0.0.0"
 edition = "2018"
 
-# Regenerate: $ CARGO_BAZEL_REPIN=1 bazelisk build //...
-# Reference: @crate_index//:proxy-wasm (etc)
+# Regenerate: $ bazelisk run bazel/cargo:crates_vendor -- --repin
+# Reference: //bazel/cargo/remote:proxy-wasm (etc)
 [dependencies]
-proxy-wasm = "0.2.1"
-log = "0.4"
-regex = "1.9"
-url = "2.4"
+proxy-wasm = "=0.2.1"
+log = "=0.4"
+regex = "=1.9"
+url = "=2.4"
```

Comment:
Can we roll back the addition of the "=" prefixes, or is it needed for some reason?

Reply:
Cargo will sometimes use the newer version of a crate if there is one available. The "=" prefix tells cargo to only use the version specified.

Reply:
Is repinning to later patch versions a bad thing? The last known working dependency tree is captured in the generated build files, so unless you regen those with --repin there is no change. I would prefer not "hard pinning" if possible.

Reply:
I've already ran into some cases where repining the latest version has resulted in plugin samples breaking. Also, after repining there isn't an easy way to rollback when a breakage would occur. I think it's better to "hard pin" these top level dependencies, otherwise when adding a new dependency and repining you could unintentionally break other plugins since they now have "updated" dependencies. I think if we weren't Wasm based it would make more sense to use the latest version of libraries, but because we are we want to avoid pulling in changes that fundamentally don't work for us(threading, IO, etc.). Rust/Cargo will still recognize a crate as stable or healthy when these changes occur and pull in changes that silently break sample plugins.

Reply:
> I've already ran into some cases where repining the latest version has resulted in plugin samples breaking

Can you elaborate on what sort of breakages you saw? If our samples break with patch version updates to dependencies, I'm wondering if there might be an underlying issue we need to fix. Maybe I'm missing something, but ISTM that fixing this by pinning to exact versions is addressing a symptom rather than the cause.

Reply:
After an offline discussion, I agree that we should handle breakages as they occur and only pin dependencies as needed as a temporary solution.